src/components/video_player.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * This component is mainly to handle the player's error and loading state nicely, being able to display something custom in those cases.
+ * The react-player component knows much more than youtube and these props, but only these are needed for now. Refer to it's documentation before changing this component.
+ */
+import * as React from 'react';
+import YouTubePlayer from 'react-player/lib/players/YouTube';
+
+import Spinner from "./spinner";
+
+export interface VideoPlayerProps {
+    /** url for the react-player component */
+    url: string;
+    /** Optional width to be given to the container and the video. Will accept values that are valid in CSS */
+    width?: string;
+    /** Optional height to be given to the container and the video. Will accept values that are valid in CSS */
+    height?: string;
+}
+
+export interface VideoPlayerState {
+    loading: boolean;
+    error: boolean;
+    wrongUrl: boolean;
+}
+
+export default class VideoPlayer extends React.Component<VideoPlayerProps, VideoPlayerState> {
+    static defaultProps = {
+        width: '100%',
+        height: '100%'
+    };
+
+    state = {
+        loading: true,
+        error: false,
+        wrongUrl: false
+    }
+
+    private _ready() {
+        this.setState({ loading: false })
+    }
+
+    private _error(e) {
+        this.setState({
+            loading: false,
+            error: true
+        });
+    }
+
+    private _playerClasses() {
+        const {loading,error,wrongUrl} = this.state;
+        if (loading || error || wrongUrl) {
+            return 'video-player--player video-player--player__hidden'
+        } else {
+            return 'video-player--player'
+        }
+    }
+
+    componentDidMount() {
+        if (!YouTubePlayer.canPlay(this.props.url)) {
```

Comment:
I couldn't get this to trigger in storybook

Reply:
It should work now.

Reply:
You mentioned that when you change the url from an invalid one to a valid one, the video does show as expected, but there is not spinner for it's loading.
I spent about a day trying to solve this issue, but could not come up with a working solution for it. The the video player component doesn't have a callback for a loading state, neither does youtube's API.
I discussed it with Simon, we couldn't come up with a solution together either.
So I suggest to leave it as it is, since this situation is not even going to happen in the introductory popup.

Reply:
I have fixed this using componentDidUpdate